pephub/routers/eido/eido.py

```diff
@@ -377,28 +113,57 @@ async def validate(
 
             p = peppy.Project(f"{dirpath}/{init_file.filename}")
 
-    # save schema string to temp file, then read in with eido
-    with tempfile.NamedTemporaryFile(mode="w") as schema_file:
-        schema_file.write(schema)
-        schema_file.flush()
+    if schema is None and schema_registry is None:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "Must supply either a registry path or a list of files to validate."
+            },
+        )
+    
+    if schema_registry is not None:
+        schema_url = f"https://schema.databio.org/{schema_registry}.yaml"
+            
         try:
-            schema_dict = eido.read_schema(schema_file.name)[0]
-        except eido.exceptions.EidoSchemaInvalidError as e:
+            response = requests.get(schema_url)
+            response.raise_for_status()  # Check if the request was successful
+            yaml_string = response.text
+        except requests.exceptions.RequestException as e:
             raise HTTPException(
-                status_code=200,
-                detail={"error": f"Schema is invalid: {str(e)}"},
+                status_code=400,
+                detail={"error": f"Error fetching schema: {str(e)}"},
```

Comment:
This is a nice approach. If I read this right... if they send a bad schema (`this/doesnt-exist`) then it returns with `400`?

Reply:
Yep, an should show up as invalid when looking at a project. I guess one question would be if they have 'No Schema' should we even show whether it's valid or not since we're not validating against anything?